src/MANYBODY/pair_eam.cpp

```diff
@@ -322,6 +326,16 @@ void PairEAM::compute(int eflag, int vflag)
     }
   }
 
+  if (eflag && (exceeded_rhomax >= 0)) {
```

Comment:
Is the intent here to only check for a possible warning on a step where energy is needed (e.g. a thermo output step)?  And then disable the check and the warning message thereafter?  I agree with that strategy
so that Allreduce is not called often. 

However, it then seems like beyond_rhomax should not be zeroed at the top of compute() every step?  Rather it should be initialized to zero in init_style()

Reply:
`beyond_rhomax` is a local variable in the compute method and that is either 0 or 1 depending on whether `rho[i]` has exceeded rhomax.

`exceeded_rhomax` is a per-class instance variable that is set to 0 at the beginning of each run and then becomes 1 during the MPI_Allreduce() when `beyond_rhomax` is 1 for at least one of the MPI ranks. If `exceeded_rhomax` is 1 after the reduction, the warning is printed and then `exceeded_rhomax` is set to -1 so it won't flag the same warning during the run again.

Reply:
Yes, the intent is to only check on steps with energy output (which is often the first step) to reduce the frequency of the MPI_Allreduce().